test/IRGen/struct_resilience.swift

```diff
@@ -189,6 +189,81 @@ public func resilientAny(s : ResilientWeakRef) {
 // CHECK:  call void @__swift_destroy_boxed_opaque_existential_0(%Any* [[ANY]])
 // CHECK:  ret void
 
+// Make sure that MemoryLayout properties access resilient types' metadata
+// instead of hardcoding sizes based on compile-time layouts.
+
+// CHECK-LABEL: define swiftcc i64 @"$s17struct_resilience38memoryLayoutDotSizeWithResilientStructSiyF"() #0
+public func memoryLayoutDotSizeWithResilientStruct() -> Int {
+  // CHECK-NEXT: entry:
+  // CHECK-NEXT: [[ZERO:%[-a-zA-Z$._0-9]*]] = call swiftcc %swift.metadata_response @"$s16resilient_struct4SizeVMa"(i64 0)
+  // CHECK-NEXT: [[ONE:%[-a-zA-Z$._0-9]*]] = {{.*}} [[ZERO]]
```

Comment:
Can you fill this in properly? You should be able to find the pattern elsewhere in the file and copy it here.

Reply:
Sure; doing it this way was probably too cute.